lasagne/tests/layers/test_noise.py

```diff
@@ -138,7 +138,7 @@ def test_get_output_for_deterministic(self, layer):
         input = theano.shared(numpy.ones((3, 3)))
         result = layer.get_output_for(input, deterministic=True)
         result_eval = result.eval()
-        assert (result_eval == input.eval()).all()
+        assert (np.array(result_eval) == np.array(input.eval())).all()
```

Comment:
I see, it returns a `GpuArray` for which `==` only checks for identity. A shorter variant would be `np.allclose(result.eval(), input.eval())`, which handles the conversion. The tests above will need to be changed as well then -- they assert that the comparison is false, but if the comparison only checks for identity, this is trivially fulfilled.

Also for some reason, the input is not floatX, I think this was not intended. It's not terribly important for the test; we can rely on Theano that CPU and GPU versions are the same (we still might want to run the test suite with float64 warnings enabled and see if there are more glitches like this). But why is the result still a gpuarray then? Ah, I see, it's a float64 GpuArray!

Phew. I think we may need to go through all the tests and see if they're using `==` or `!=` where they should use `np.allclose` instead.